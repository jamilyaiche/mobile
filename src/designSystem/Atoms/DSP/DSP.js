import React from 'react'
import { Text } from 'react-native'
import PropTypes from 'prop-types'
import Style from '../../../assets/styles/Style'

const DSP = (props) => {
  let stAlign = Style.pTextJustify
  if (props.textAlign === 'left') {
    stAlign = Style.pTextLeft
  }
  return (
        <Text
            style={[
              Style.p,
              stAlign
            ]}
        >
            {props.children}
        </Text>
  )
}

DSP.defaultProps = {
  textAlign: 'justify'
}

DSP.proptypes = {
  textAlign: PropTypes.oneOf(['justify', 'left'])
}

export default DSP
