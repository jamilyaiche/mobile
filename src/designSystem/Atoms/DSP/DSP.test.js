import React from 'react';
import renderer from 'react-test-renderer'
import DSP from "./DSP"

describe("DSP tests",()=>{
    it('test render DSP',()=>{
        const wrapper=renderer.create(<DSP></DSP>)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    it('test render DSP text',()=>{
        const wrapper=renderer.create(<DSP>test</DSP>)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    it('test render DSP text left',()=>{
        const wrapper=renderer.create(<DSP textAlign={'left'}>test</DSP>)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
})