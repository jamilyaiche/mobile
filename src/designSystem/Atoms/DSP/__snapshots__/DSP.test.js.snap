// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSP tests test render DSP 1`] = `
<Text
  style={
    Array [
      Object {
        "color": undefined,
        "fontFamily": "Montserrat-Regular",
        "fontSize": 14,
        "letterSpacing": 1,
        "lineHeight": 24,
        "marginBottom": 10,
        "marginTop": 10,
      },
      Object {
        "textAlign": "justify",
      },
    ]
  }
/>
`;

exports[`DSP tests test render DSP text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": undefined,
        "fontFamily": "Montserrat-Regular",
        "fontSize": 14,
        "letterSpacing": 1,
        "lineHeight": 24,
        "marginBottom": 10,
        "marginTop": 10,
      },
      Object {
        "textAlign": "justify",
      },
    ]
  }
>
  test
</Text>
`;

exports[`DSP tests test render DSP text left 1`] = `
<Text
  style={
    Array [
      Object {
        "color": undefined,
        "fontFamily": "Montserrat-Regular",
        "fontSize": 14,
        "letterSpacing": 1,
        "lineHeight": 24,
        "marginBottom": 10,
        "marginTop": 10,
      },
      Object {
        "textAlign": "left",
      },
    ]
  }
>
  test
</Text>
`;
